### PR TITLE
Init buidler starter

### PR DESCRIPTION
**Pull request by nmlinaric**
Initial config for buidler starter repo.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    "mocha": "^8.0.1",
+    "solhint": "^3.0.0",
+    "solidity-coverage": "^0.7.9",
+    "ts-generator": "^0.0.8",
+    "ts-node": "^8.10.2",
+    "typechain": "^2.0.0",
+    "typescript": "^3.9.6"
+  },
+  "dependencies": {
+    "ethers": "^5.0.5"
```
Comment (mpetrunic) [inline package.json]: dev dep

File: package.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    "typechain": "npx buidler typechain"
+  },
+  "devDependencies": {
+    "@nodefactory/buidler-typechain": "^0.1.0",
+    "@nodefactory/env-prompt": "^1.0.8",
+    "@nomiclabs/buidler": "1.3.5",
+    "@nomiclabs/buidler-ethers": "^2.0.0",
+    "@nomiclabs/buidler-solhint": "^1.3.3",
+    "@nomiclabs/buidler-waffle": "^2.0.0",
+    "@openzeppelin/contracts": "^3.1.0",
```
Comment (mpetrunic) [inline package.json]: not a dev dep

File: package.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    "@typechain/ethers-v4": "^1.0.0",
```
Comment (mpetrunic) [inline package.json]: https://www.npmjs.com/package/@typechain/ethers-v5

File: tslint.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (mpetrunic) [inline tslint.json]: please use eslint and not tslint(deprecated)

Comment (mpetrunic) [reply tslint.json]: add prettier as eslint plugin as well

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+  "compilerOptions": {
+    "target": "es6",
```
Comment (mpetrunic) [inline tsconfig.json]: es2018

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: wait-for-it.sh [hunk 6]
```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env bash
```
Comment (morrigan) [inline wait-for-it.sh:1]: This script actually doesn't work for mac so would be cool to fix it or replace it by something else:
`timeout: command not found`

File: testingData/Accounts.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
+const accounts = [
```
Comment (morrigan) [inline testingData/Accounts.ts]: can this be inside `test` dir?

File: README.md [hunk 8]
```diff
@@ -1 +1,31 @@
-# Buidler-Starter
\ No newline at end of file
+# Buidler-Starter
+
+## Requirements
+
+Following software is required to be installed to use this repo:
+* [NodeJS](https://nodejs.org/en/) >= v12.0.0
+* [Nomic Labs buidler](https://buidler.dev/getting-started/#overview)
```
Comment (morrigan) [inline README.md]: isn't this and deploy part of the dependencies, so they don't have to be installed separately?

Comment (nmlinaric) [reply README.md]: Yes, but wanted to mention that those two are used, will change `readme` to something more appropriate 🙂

File: wait-for-it.sh (hunk 6 above)
Comment (nmlinaric) [reply wait-for-it.sh:1]: @morrigan Would it be ok to put [this](https://github.com/vishnubob/wait-for-it/issues/15) workaround in `readme` if using macOS? Or should I search for an alternative?

Comment (morrigan) [reply wait-for-it.sh:1]: I guess that's fine

**Review by morrigan — APPROVED** [review]
👍

**Pull request merged**